src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double smallBase;
+    private double bigBase;
+    private double side;
+
+    public IsoscelesTrapezoid() {
+    }
+
+    public IsoscelesTrapezoid(String color, double base1, double base2, double side) {
+        this.setColor(color);
+        this.smallBase = Math.min(base1, base2);
+        this.bigBase = Math.max(base1, base2);
+        this.side = side;
+    }
+
+    @Override
+    public double getArea() {
+        double height = Math.sqrt(side * side - Math.pow((bigBase - smallBase) / 2, 2));
+        return (smallBase + bigBase) * height / 2;
+    }
+
+    @Override
+    public void draw() {
+        StringBuilder info = new StringBuilder();
+        info.append("Figure: isosceles trapezoid, area: ")
+                .append(getArea()).append(" sq.units, small base: ")
+                .append(smallBase).append(" units, big base: ")
+                .append(bigBase).append(" units, side: ")
+                .append(side).append(" units, color: ")
+                .append(getColor());
+        System.out.println(info);
+    }
+
+    @Override
+    public Figure getRandomFigure() {
```

Comment:
please move this logic to FigureSupplier class, method getRandomFigure()

same for all child figures
you can use switch case based on random number - to determine which figure should be created in  getRandomFigure() method of FigureSupplier class